internal/storage/dummy_io_bucket.go

```diff
@@ -292,3 +296,55 @@ func (dr *dummyReader) Close() error {
 func (dr *dummyReader) ReadHandle() storagev2.ReadHandle {
 	return dr.readHandle
 }
+
+////////////////////////////////////////////////////////////////////////
+// dummyMultiRangeDownloader
+////////////////////////////////////////////////////////////////////////
+
+type dummyMultiRangeDownloader struct {
+	perMBLatency time.Duration
+	wg           sync.WaitGroup
+}
+
+// zeroReader is an io.Reader that always reads zeros.
+type zeroReader struct{}
+
+func (zeroReader) Read(p []byte) (int, error) {
+	clear(p)
+	return len(p), nil
+}
+
+func (d *dummyMultiRangeDownloader) Add(output io.Writer, offset, length int64, callback func(int64, int64, error)) {
+	d.wg.Add(1)
+	go func() {
+		defer d.wg.Done()
+
+		// Simulate latency
+		time.Sleep(calculateLatency(length, d.perMBLatency))
+
+		// Write zeros
+		// output writer is bytes.Buffer which implements io.ReaderFrom interface
+		bytesWritten, err := io.Copy(output, io.LimitReader(zeroReader{}, length))
```

Comment:
Should we re-use dummyReader only instead of zeroReader - we can also utilize the latency simulation?